TestResultSummaryService/routes/getTestBuildsByMachine.js

```diff
@@ -0,0 +1,34 @@
+const { query } = require('winston');
+const { TestResultsDB, ObjectID } = require('../Database');
+module.exports = async (req, res) => {
+    let { buildNameRegex, machineRegex } = req.query;
+    const db = new TestResultsDB();
+
+    let query = {};
+    if (buildNameRegex) {
+        query.buildName = { $regex: buildNameRegex };
+    }
+    if (machineRegex) {
+        query.machine = { $regex: machineRegex };
+    }
+
+    let result = await db.aggregate([
+        {
+            $match: query,
+        },
+        {
+            $project: {
+                buildName: '$buildName',
+                machine: '$machine',
+                passRate: {
+                    $concat: [
+                        { $substr: ['$testSummary.passed', 0, -1] },
+                        '/',
+                        { $substr: ['$testSummary.executed', 0, -1] },
```

Comment:
In this case, we need to check `buildResult`, not `testSummary`. The passRate is the number of the test builds that have `buildResult` = SUCCESS / the number of all test builds